pageserver/src/http/routes.rs

```diff
@@ -1384,7 +1387,18 @@ async fn put_tenant_location_config_handler(
         tracing::info!("No flush requested when configuring");
     }
 
-    json_response(StatusCode::OK, ())
+    // This API returns a vector of pageservers where the tenant is attached: this is
+    // primarily for use in the sharding service.  For compatibilty, we also return this
+    // when called directly on a pageserver, but the payload is always zero or one shards.
+    let mut response = TenantLocationConfigResponse { shards: Vec::new() };
+    if attached {
+        response.shards.push(TenantShardLocation {
+            shard_id: tenant_shard_id,
+            node_id: state.conf.id,
+        })
+    }
+
+    json_response(StatusCode::OK, response)
```

Comment:
Yep.  This is reflected in the swagger + there was a thread with control plane folks about it -- this exists so that they can always attach a compute right after calling this API, without having to teach control plane to wait for a notification to their hook (a la https://github.com/neondatabase/neon/pull/6471)